routing/car_router.cpp

```diff
@@ -196,41 +197,86 @@ class OSRMRoutingResult : public turns::IRoutingResult
   RoutingMapping & m_routingMapping;
 };
 
+bool FindSingleOsrmRoute(FeatureGraphNode const & source, FeatureGraphNode const & target,
+                         RouterDelegate const & delegate, TRoutingMappingPtr & mapping, Index const & index,
+                         Route & mwmRoute)
+{
+  vector<Junction> mwmRouteGeometry;
+  Route::TTurns mwmTurns;
+  Route::TTimes mwmTimes;
+  Route::TStreets mwmStreets;
+
+  LOG(LINFO, ("OSRM route from", MercatorBounds::ToLatLon(source.segmentPoint),
+              "to", MercatorBounds::ToLatLon(target.segmentPoint)));
+
+  RawRoutingResult routingResult;
+  if (!FindSingleRoute(source, target, mapping->m_dataFacade, routingResult))
+    return false;
+
+  OSRMRoutingResult resultGraph(index, *mapping, routingResult);
+  if (MakeTurnAnnotation(resultGraph, delegate, mwmRouteGeometry, mwmTurns, mwmTimes, mwmStreets)
+      != routing::IRouter::NoError)
+  {
+    LOG(LWARNING, ("Can't load road path data from disk for", mapping->GetCountryName()));
+    return false;
+  }
+
+  mwmRoute.SetTurnInstructions(move(mwmTurns));
+  mwmRoute.SetSectionTimes(move(mwmTimes));
+  mwmRoute.SetStreetNames(move(mwmStreets));
+
+  vector<m2::PointD> mwmPoints;
+  JunctionsToPoints(mwmRouteGeometry, mwmPoints);
+  mwmRoute.SetGeometry(mwmPoints.cbegin(), mwmPoints.cend());
+
+  return true;
+}
+} //  namespace
+
 // static
-bool OsrmRouter::CheckRoutingAbility(m2::PointD const & startPoint, m2::PointD const & finalPoint,
-                                     TCountryFileFn const & countryFileFn, Index * index)
+bool CarRouter::CheckRoutingAbility(m2::PointD const & startPoint, m2::PointD const & finalPoint,
+                                    TCountryFileFn const & countryFileFn, Index * index)
 {
   RoutingIndexManager manager(countryFileFn, *index);
   return manager.GetMappingByPoint(startPoint)->IsValid() &&
          manager.GetMappingByPoint(finalPoint)->IsValid();
 }
 
-OsrmRouter::OsrmRouter(Index * index, TCountryFileFn const & countryFileFn)
-    : m_pIndex(index), m_indexManager(countryFileFn, *index)
+CarRouter::CarRouter(Index * index, TCountryFileFn const & countryFileFn,
+                     unique_ptr<IRouter> roadGraphRouter)
+    : m_pIndex(index), m_indexManager(countryFileFn, *index), m_aStarRouter(move(roadGraphRouter))
```

Comment:
Sorry, but m_aStarRouter is not a good name, as you receiving IRouter. Please, rename to m_router.